modules/thunder_article/src/Form/ThunderNodeForm.php

```diff
@@ -0,0 +1,139 @@
+      $container->get('entity_type.bundle.info'),
+      $container->get('datetime.time'),
+      $container->get('current_user'),
+      $container->has('content_moderation.moderation_information') ? $container->get('content_moderation.moderation_information') : NULL
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function form(array $form, FormStateInterface $form_state) {
+    /** @var \Drupal\Core\Entity\ContentEntityFormInterface $form_object */
+    $form_object = $form_state->getFormObject();
+    /** @var \Drupal\node\NodeInterface $entity */
+    $entity = $form_object->getEntity();
+
+    if ($this->moderationInfo->hasPendingRevision($entity)) {
+      $user = $this->entityTypeManager->getStorage('user')->load($entity->getRevisionUserId());
+      $this->messenger()->addWarning($this->t('This %entity_type has unpublished changes from user %user.', ['%entity_type' => $entity->get('type')->entity->label(), '%user' => $user->label()]));
+    }
+
+    return parent::form($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function actions(array $form, FormStateInterface $form_state) {
+    $element = parent::actions($form, $form_state);
+
+    /** @var \Drupal\Core\Entity\ContentEntityFormInterface $form_object */
+    $form_object = $form_state->getFormObject();
+    /** @var \Drupal\node\NodeInterface $entity */
+    $entity = $form_object->getEntity();
+
+    if (!$this->moderationInfo || !$this->moderationInfo->isModeratedEntity($entity)) {
+      return $element;
+    }
+
+    $element['status'] = [
+      '#type' => 'item',
+      '#markup' => $this->entity->isNew() || !$this->moderationInfo->isDefaultRevisionPublished($entity) ? $this->t('of unpublished @entity_type', ['@entity_type' => strtolower($entity->type->entity->label())]) : $this->t('of published @entity_type', ['@entity_type' => strtolower($entity->type->entity->label())]),
+      '#weight' => 200,
+      '#wrapper_attributes' => [
+        'class' => ['status'],
+      ],
+    ];
+
+    $state = $this->moderationInfo->getWorkflowForEntity($entity)->getTypePlugin()->getState($entity->moderation_state->value);
+    $element['moderation_state_current'] = [
+      '#type' => 'item',
+      '#markup' => $state->label(),
+      '#weight' => 210,
+      '#wrapper_attributes' => [
+        'class' => ['status', $state->id()],
+      ],
+    ];
+
+    if ($this->moderationInfo->hasPendingRevision($entity)) {
+      $route_info = Url::fromRoute('node.revision_delete_confirm', [
+        'node' => $entity->id(),
+        'node_revision' => $entity->getLoadedRevisionId(),
+      ]);
+      if ($this->getRequest()->query->has('destination')) {
+        $query = $route_info->getOption('query');
+        $query['destination'] = $this->getRequest()->query->get('destination');
+        $route_info->setOption('query', $query);
+      }
+      $element['delete_revision'] = [
+        '#type' => 'link',
+        '#title' => $this->t('Delete this version'),
```

Comment:
I'm guessing that main purpose here is to allow users to start from the published revision again. Have we considered offering the ability to create a new draft based on the published revision?

Reply:
The purpose here is to be able to quickly dismiss the current draft, not necessarily to create a new one. But the use case of creating a new draft (that overrides an existing one) might be interesting as well, I will discuss this when we present the current state to our stake holders.

Reply:
But doesn't this get into misleading territory when there are multiple forward revisions? In a revisioning world the ideal thing is to not to do deletes.

Reply:
Yes, that is a problem.

Reply:
The current idea is to change this to something like "Revert to last published version"

Reply:
👍 That sounds like a good idea because:
* Non destructive
* Less deletes all in a row in the UI.